### PR TITLE
refactor: neuron detail modals custom event

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

Use `CustomEvent` instead of context store to trigger the opening of the modal on the neuron detail pages.

# Changes

- use `CustomEvent` to open modal
- replace and remove `modal` and additional info from Nns and Sns context (those that have been added this week in previous PRs)
- update `<EditFollowNeurons />` to get `neuron` variable from `definedNeuronsStore` (see notes)

# Notes

- Nns vs Sns approaches are a bit different because Nns neuron store is global while Sns neuron is tight to context. Long story short, in Nns the data are pass within the modal and in Sns only the type of modal to open is throw, the rest being tight to the context store.

- To follows neuron, two modals are displayed. One that list the followes and one that is used to add or remove follows. The second one update the neuron. Since these information are pass through custom event, the source modal that list the followee was not automatically updated anymore because not tight with a store. Since addiing a follow refresh the global store, the modal that list can be linked with `definedNeuronsStore`

### CODE REVIEW

File: frontend/src/lib/modals/neurons/FollowNeuronsModal.svelte [hunk 1]
```diff
@@ -1,15 +1,15 @@
 <script lang="ts">
-  import type { NeuronInfo } from "@dfinity/nns";
+  import type { NeuronId } from "@dfinity/nns";
   import EditFollowNeurons from "$lib/components/neurons/EditFollowNeurons.svelte";
   import { i18n } from "$lib/stores/i18n";
   import { Modal } from "@dfinity/gix-components";
 
-  export let neuron: NeuronInfo;
+  export let neuronId: NeuronId;
```
Comment (lmuntaner) [inline frontend/src/lib/modals/neurons/FollowNeuronsModal.svelte:7]: Why this change?

File: frontend/src/lib/modals/neurons/NnsNeuronModals.svelte [hunk 2]
```diff
@@ -20,79 +14,80 @@
...
 
-  {#if modal === "join-community-fund"}
+  {#if type === "join-community-fund"}
     <JoinCommunityFundModal on:nnsClose={close} {neuron} />
   {/if}
 
-  {#if modal === "follow"}
-    <FollowNeuronsModal on:nnsClose={close} {neuron} />
+  {#if type === "follow"}
+    <FollowNeuronsModal on:nnsClose={close} neuronId={neuron.neuronId} />
```
Comment (lmuntaner) [inline frontend/src/lib/modals/neurons/NnsNeuronModals.svelte:83]: Why not passing the whole neuron?

File: frontend/src/lib/modals/neurons/NnsNeuronModals.svelte [hunk 3]
```diff
@@ -20,79 +14,80 @@
...
     <AddHotkeyModal on:nnsClose={close} {neuron} />
   {/if}
 
-  {#if modal === "voting-history" && selectedFollowee !== undefined}
-    <VotingHistoryModal
-      neuronId={selectedFollowee.neuronId}
-      on:nnsClose={close}
-    />
+  {#if type === "voting-history" && followee !== undefined}
+    <VotingHistoryModal neuronId={followee.neuronId} on:nnsClose={close} />
```
Comment (lmuntaner) [inline frontend/src/lib/modals/neurons/NnsNeuronModals.svelte:91]: Should we move this modal outside of the `if nonNullish(neuron)`? It doesn't need the neuron.

File: frontend/src/lib/modals/neurons/StakeNeuronModal.svelte [hunk 4]
```diff
@@ -199,8 +199,8 @@
     {/if}
   {/if}
   {#if currentStep?.name === "EditFollowNeurons"}
-    {#if newNeuron !== undefined}
-      <EditFollowNeurons neuron={newNeuron} />
+    {#if newNeuron !== undefined && newNeuronId !== undefined}
+      <EditFollowNeurons neuronId={newNeuronId} />
```
Comment (lmuntaner) [inline frontend/src/lib/modals/neurons/StakeNeuronModal.svelte:203]: Ditto on why the change

**Review by lmuntaner — COMMENTED** [review]
Looking good!

Just a question on a change I don't fully understand.

Thanks!

File: frontend/src/lib/modals/neurons/FollowNeuronsModal.svelte (hunk 1 above)
Comment (peterpeterparker) [reply frontend/src/lib/modals/neurons/FollowNeuronsModal.svelte:7]: I tried to explain that in the "Notes" of the description of the PR 😉

> To follows neuron, two modals are displayed. One that list the followes and one that is used to add or remove follows. The second one update the neuron. Since these information are pass through custom event, the source modal that list the followee was not automatically updated anymore because not tight with a store. Since addiing a follow refresh the global store, the modal that list can be linked with definedNeuronsStore

So `FollowNeuronsModal` uses `EditFollowNeurons` that needs to fetch / sync the neuron from `$definedNeuronsStore` instead of receiving the neuron as property because if the neuron is passed / initialized with the Custom Event, it is not dynamic but a one shot value.

Is that a bit more clear?

File: frontend/src/lib/modals/neurons/NnsNeuronModals.svelte (hunk 2 above)
Comment (peterpeterparker) [reply frontend/src/lib/modals/neurons/NnsNeuronModals.svelte:83]: See previous comment https://github.com/dfinity/nns-dapp/pull/1602#discussion_r1036813428

File: frontend/src/lib/modals/neurons/NnsNeuronModals.svelte (hunk 3 above)
Comment (peterpeterparker) [reply frontend/src/lib/modals/neurons/NnsNeuronModals.svelte]: Sure

File: frontend/src/lib/modals/neurons/StakeNeuronModal.svelte (hunk 4 above)
Comment (peterpeterparker) [reply frontend/src/lib/modals/neurons/StakeNeuronModal.svelte:203]: Neuron check remove

Why the change see https://github.com/dfinity/nns-dapp/pull/1602#discussion_r1036813428

File: frontend/src/lib/modals/neurons/FollowNeuronsModal.svelte (hunk 1 above)
Comment (lmuntaner) [reply frontend/src/lib/modals/neurons/FollowNeuronsModal.svelte:7]: got it, yes. Thanks!

**Comment by peterpeterparker** [comment]
Haha no worries @lmuntaner, thanks for the review!
